seismicpro/refractor_velocity/utils.py

```diff
@@ -91,3 +92,64 @@ def dump_refractor_velocities(refractor_velocities, path, encoding="UTF-8"):
         data[i] = [*rv.coords.names] + [*rv.coords.coords] + list(rv.params.values())
     df = pd.DataFrame(data, columns=columns).convert_dtypes()
     df.to_string(buf=path, float_format=lambda x: f"{x:.2f}", index=False, encoding=encoding)
+
+
+def reduce_offsets_and_times(survey, first_breaks_col=HDR_FIRST_BREAK, reduce_step=20):
+    """Reduce the offsets and times of first breaks stored in the survey headers.
+
+    Method splits the survey headers by bins with the same size define by `reduce_step` using offsets values and
+    calculate the mean separately by bins. The first breaks times uses same bins and also calculate the mean
+    separately by bins.
+    """
+    headers = survey.headers[['offset', first_breaks_col]].copy()
+    headers['bins'] = headers['offset'] // reduce_step
+    reduced_headers = headers.groupby(by='bins', sort=False).mean()
+    return reduced_headers['offset'].to_numpy(), reduced_headers[first_breaks_col].to_numpy()
```

Comment:
No need to create intermediate column
```suggestion
    return headers.groupby(headers['offset'] // reduce_step, sort=False).mean().values.T
```